src/grt/grt-wave_opt_file.adb

```diff
@@ -42,21 +43,27 @@ package body Grt.Wave_Opt_File is
          when Warning =>
             Report_C ("warning: ");
       end case;
-      Report_C ("in file '");
       Report_C (File_Path.all);
-      Report_C ("' at line ");
-      To_String (Lineno_Str, First, Ghdl_I32 (Line_Context.Num));
-      Report_C (Lineno_Str (First .. Lineno_Str'Last));
-      Report_C (" - ");
-      Report_C (Line_Context.Str.all);
-      Report_C (" : ");
+      Report_C (":");
+      To_String (Value_Str, First, Ghdl_I32 (Line_Pos));
```

Comment:
No need to convert: there is a Report_C for Natural, so:
  Report_C (Line_Pos);
should work